contracts/axelarnet-gateway/src/contract/execute.rs

```diff
@@ -0,0 +1,134 @@
+use axelar_wasm_std::nonempty;
+use cosmwasm_std::{to_json_binary, Addr, Api, HexBinary, Response, Storage, WasmMsg};
+use error_stack::{report, Result, ResultExt};
+use router_api::client::Router;
+use router_api::{Address, ChainName, CrossChainId, Message};
+use sha3::{Digest, Keccak256};
+
+use crate::contract::Error;
+use crate::events::AxelarnetGatewayEvent;
+use crate::msg::AxelarExecutableExecuteMsg;
+use crate::state::{self};
+
+// TODO: Retrieve the actual tx hash from core, since cosmwasm doesn't provide it. Use a placeholder in the meantime.
+const PLACEHOLDER_TX_HASH: &str =
+    "0x0000000000000000000000000000000000000000000000000000000000000000";
+
+pub(crate) fn call_contract(
+    store: &mut dyn Storage,
+    router: &Router,
+    chain_name: ChainName,
+    sender: Addr,
+    destination_chain: ChainName,
+    destination_address: Address,
+    payload: HexBinary,
+) -> Result<Response, Error> {
+    let counter =
+        state::increment_message_counter(store).change_context(Error::InvalidStoreAccess)?;
+
+    let cc_id = CrossChainId {
+        source_chain: chain_name.into(),
+        message_id: nonempty::String::try_from(format!("{0}-{1}", PLACEHOLDER_TX_HASH, counter))
+            .change_context(Error::MessageIdConstructionFailed)?,
+    };
+
+    let payload_hash = Keccak256::digest(payload.as_slice()).into();
+
+    let msg = Message {
+        cc_id: cc_id.clone(),
+        source_address: Address::try_from(sender.clone().into_string())
+            .change_context(Error::InvalidSender(sender))?,
```

Comment:
in what situation would this ever error? Seems like it should be an asset or expect

Reply:
sure, i thought we were always using errors instead of unwrapping